### PR TITLE
[2.4.0]Fix/1973 fix doublon in simple search backend

### PR DESCRIPTION
**Pull request by inkhey**
related to #1973

## Checkpoints

The goal of this section is to help code reviewer to do required checks on this pull request. Please don't edit this list.
If one or more of these checkpoints are out of scope, please write below why they are.

- [x] Code is clear enough
- [x] If there are FIXME in the code, then there are associated issues and issue is referenced in the FIXME
- [x] If there are TODO, NOTE or HACK in code, date and developer initials are present
- [x] Automated tests have been written (covering feature or non regression), or *at least* an issue has been created for test implementation

### CODE REVIEW

File: backend/tracim_backend/lib/core/content.py [hunk 1]
```diff
@@ -2158,8 +2158,8 @@ def search(
                 # INFO - G.M - 2019-06-13 - avoid duplication of same content in result list
                 continue
             if current_offset >= offset:
-                parsed_content_ids.append(content.content_id)
                 results.append(content)
+            parsed_content_ids.append(content.content_id)
             current_offset += 1
 
         return results, current_offset
```
Comment (lebouquetin) [inline backend/tracim_backend/lib/core/content.py]: I see now that we return a tuple. We must avoid this kind of behavior which is hard to understand: the typing help understanding the type of data, but understand what the data is requires to read the code.

Comment (inkhey) [reply backend/tracim_backend/lib/core/content.py]: code changed in order to don't need them.

File: backend/tracim_backend/lib/core/content.py [hunk 2]
```diff
@@ -49,6 +49,7 @@
 from tracim_backend.exceptions import WorkspacesDoNotMatch
 from tracim_backend.lib.core.notifications import NotifierFactory
 from tracim_backend.lib.core.userworkspace import RoleApi
+from tracim_backend.lib.search.models import SimpleContentSearchResponse
```
Comment (lebouquetin) [inline backend/tracim_backend/lib/core/content.py]: I suggest to rename `SimpleContentSearchResponse` to `SimpleContentSearchResult`?

Comment (lebouquetin) [reply backend/tracim_backend/lib/core/content.py]: By the way, the search feature must be moved to `lib.search.xxx` module. There is no reason for `lib.core.content` having a dependency on `lib.search.models`.

What you should do is:

- have a *simple search* module which has a dependency on `lib.core.content` and which implements the search method.
- have a dependency on `lib.search.models` in *simple search* module
- have a `lib.core.content` module which does not known anything about search.

Comment (inkhey) [reply backend/tracim_backend/lib/core/content.py]: Is the new implementation ok ?

**Comment by lebouquetin** [comment]
@inkhey if only copy/paste, then split clearly core / simple / elasticsearch parts of the search before we merge

**Pull request merged**